include/circt/Dialect/StaticLogic/StaticLogic.td

```diff
@@ -72,4 +72,137 @@ def ReturnOp : Op<StaticLogic_Dialect, "return", [Terminator]> {
   let arguments = (ins Variadic<AnyType>: $operands);
 }
 
+def PipelineWhileOp : Op<StaticLogic_Dialect, "pipeline.while", []> {
+  let summary = "StaticLogic dialect pipeline while-loop.";
+  let description = [{
+    The `staticlogic.pipeline.while` operation represents a statically scheduled
+    pipeline stucture that executes while a condition is true. For more details,
+    see: https://llvm.discourse.group/t/rfc-representing-pipelined-loops/4171.
+
+    This is the top-level operation representing a high-level pipeline. It is
+    not isolated from above, but could be if this is helpful. A pipeline
+    contains two regions: `condition` and `stages`.
+
+    The pipeline may accept an optional `iter_args`, similar to the SCF dialect,
+    for representing loop-carried values like induction variables or reductions.
+    When the pipeline starts execution, the registers indicated as `iter_args`
+    by `staticlogic.pipeline.terminator` should be initialized to the initial
+    values specified in the `iter_args` section here.
+
+    The single-block `condition` region dictates the condition under which the
+    pipeline should execute. It has a `staticlogic.yield` terminator, and the
+    pipeline executes while the yielded value is `true : i1`. It may access SSA
+    values dominating the pipeline, as well as `iter_args`, which are block
+    arguments.
```

Comment:
Do we expect/enforce the condition region to be combinatorial? I think that would be reasonable limitation, but then should be mentioned here.

Reply:
Yes, that was my thinking. I'll document this, and the verifier can enforce it.